set_delivery_address/models/sale_order.py

```diff
@@ -0,0 +1,31 @@
+from odoo import models, api
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        super(SaleOrder, self).onchange_partner_id()
+        list_partner = []
+        for record in self.partner_id:
+            for child in record.child_ids:
+                if child.address_verified:
+                    list_partner.append(child.name)
+
+        for record in self.partner_id:
+            for child in record.child_ids:
+                if len(list_partner) > 1:
+                    if child.name == list_partner[0]:
+                        values = {
+                            'partner_invoice_id': child.id,
+                            'partner_shipping_id': child.id,
+                        }
+                        self.update(values)
+                else:
+                    if child.address_verified:
+                        values = {
+                            'partner_invoice_id': child.id,
+                            'partner_shipping_id': child.id,
+                        }
+                        self.update(values)
```

Comment:
`list_partner = [child for record in self.partner_id for child in record.child_ids if child.address_verified]`
`if len(list_partner) > 0:`
    ` values = {`
      `  'partner_invoice_id': list_partner[0].id,`
       ` 'partner_shipping_id': list_partner[0].id,`
    `}`
    `self.update(values)`